### PR TITLE
chore: improve save/load methods for encrypted data-frames

### PR DESCRIPTION
**Pull request by RomanBredehoft**
this should make the save/load methods faster

### CODE REVIEW

File: src/concrete/ml/pandas/_utils.py [hunk 1]
```diff
@@ -133,28 +133,28 @@ def deserialize_elementwise(array: numpy.ndarray) -> numpy.ndarray:
     return numpy.vectorize(deserialize_value)(array)
 
 
-def serialize_evaluation_keys(evaluation_keys: fhe.EvaluationKeys) -> str:
+def serialize_evaluation_keys(evaluation_keys: fhe.EvaluationKeys) -> bytes:
```
Comment (andrei-stoian-zama) [inline src/concrete/ml/pandas/_utils.py:136]: why do you need hex? it requires processing that is slow

File: src/concrete/ml/pandas/dataframe.py [hunk 2]
```diff
@@ -318,9 +320,16 @@ def save(self, path: Union[Path, str]):
...
-        with path.open("w", encoding="utf-8") as file:
-            json.dump(encrypted_df_dict, file)
+        if path.suffix != ".zip":
+            path = path.with_suffix(".zip")
+
+        encrypted_df_dict, evaluation_keys = self._to_dict_and_eval_keys()
+
+        encrypted_df_json_bytes = json.dumps(encrypted_df_dict).encode(encoding="utf-8")
+
+        with zipfile.ZipFile(path, "w") as zip_file:
```
Comment (andrei-stoian-zama) [inline src/concrete/ml/pandas/dataframe.py:330]: can you make sure there is no compression. it's useless to compress the eval key which looks random and it just slows down the saving process

**Review by andrei-stoian-zama — CHANGES_REQUESTED** [review]
Thanks for the quick PR. Can you clarify a few things please?

File: src/concrete/ml/pandas/_utils.py (hunk 1 above)
Comment (RomanBredehoft) [reply src/concrete/ml/pandas/_utils.py:136]: I removed it actually, just forgot to update the docstring as well good catch

initially it was for making it serializable in json

File: src/concrete/ml/pandas/dataframe.py (hunk 2 above)
Comment (RomanBredehoft) [reply src/concrete/ml/pandas/dataframe.py]: yes

**Review by andrei-stoian-zama — APPROVED** [review]

**Pull request merged**